### PR TITLE
Truncate long traces

### PR DESCRIPTION
**Pull request by klauspost**
## Motivation and Context

Prevent excessively long request traces

## How to test this PR?

Observe traces.

## Types of changes
- [x] Optimization (provides speedup with no functional changes)

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: internal/grid/trace.go [hunk 1]
```diff
@@ -129,7 +129,11 @@ func (c *muxClient) traceRoundtrip(ctx context.Context, t *tracer, h HandlerID,
 		trace.HTTP.ReqInfo.Path = trace.Path
 	} else if v != nil {
 		// Print exported fields as single request to path.
-		trace.Path = fmt.Sprintf("%s?req=%s", trace.Path, url.QueryEscape(fmt.Sprintf("%+v", v)))
+		obj := fmt.Sprintf("%+v", v)
+		if len(obj) > 1024 {
```
Comment (harshavardhana) [inline internal/grid/trace.go:133]:
```suggestion
		if len(obj) > 1024 {
```

still 1k is a lot? @klauspost

Comment (klauspost) [reply internal/grid/trace.go:133]: It is only internal traces... So I think it is safe to say that it is rarely enabled.... What would you like it to be?

Comment (harshavardhana) [reply internal/grid/trace.go:133]: I am thinking that can this be just like 64bytes and not more than that. Seeing entire fileinfo is not a requirement really TBH.

Comment (harshavardhana) [inline internal/grid/trace.go:133]:
```suggestion
		if len(obj) > 64 {
```

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
Let's take this in ..

**Pull request merged**